incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/duplicated/MetricsLayout.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.exporter.otlp.internal.duplicated;
+
+import static org.agrona.IoUtil.unmap;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.LongConsumer;
+import java.util.function.LongSupplier;
+import java.util.stream.StreamSupport;
+
+import org.agrona.BitUtil;
+import org.agrona.concurrent.AtomicBuffer;
+
+public abstract class MetricsLayout extends Layout
+{
+    protected static final int FIELD_SIZE = BitUtil.SIZE_OF_LONG;
+    protected static final int BINDING_ID_OFFSET = 0;
+    protected static final int METRIC_ID_OFFSET = 1 * FIELD_SIZE;
+    protected static final int VALUE_OFFSET = 2 * FIELD_SIZE;
+    protected static final int NOT_FOUND = -1;
+
+    protected final AtomicBuffer buffer;
+
+    protected MetricsLayout(
+        AtomicBuffer buffer)
+    {
+        this.buffer = buffer;
+    }
+
+    @Override
+    public void close()
+    {
+        unmap(buffer.byteBuffer());
+    }
+
+    public long[][] getIds()
+    {
+        Spliterator<long[]> spliterator = Spliterators.spliteratorUnknownSize(new MetricsIterator(), 0);
+        return StreamSupport.stream(spliterator, false).toArray(long[][]::new);
```

Comment:
This is _not_ called repetitively, right? (allocation)

Reply:
It is called only when we initiate the `MetricsProcessor` (renamed to `MetricsReader` in PR #279). We need to call it 3 times (for counters, gauges and histograms), because they are in their respective data files and we need to get the id's of all.

Reply:
Okay, understood it is only called during initialization.